### ISSUE TITLE
Use newline instead of vertical tab in commit message

### ISSUE THREAD

**Issue by huntran-clgx** [gitextensions/gitextensions#10409]
### Environment

- GitExtensions version: 4.0.0
- GIT version: 2.38.1
- OS version: all
- .NET version: 6.0.1

### Issue description

When I put a new line in commit message, git extensions convert it to a vertical tab character (ASCII decimal character code 11)
This messed up with the display of commit message on other git clients, and also on Github (showed up as a question mark character)

### Steps to reproduce

Input multiple lines commit message, then check it on github

**Comment by pmiossec** [comment]
Screenshots?

**Comment by huntran-clgx** [comment]
![Commit message as shown on GitHub](https://i.imgur.com/qHFYhHk.png)

Here is the screenshot of the issue

**Comment by mstv** [comment]
Cannot reproduce:
https://github.com/pmiossec/gitextensions/commit/312eb7f1b2d9c4f1a30e517cfd5ad949dd27541d
https://github.com/mstv/gitextensions/commit/a1e6c26885f285e1ecf1588720010f7b93db43b5

**Comment by pmiossec** [comment]
Is it possible that you have a commit hook that causes this behavior?
Because I have never encountered such behavior from GitExtensions...

**Comment by pmiossec** [comment]
Maybe try to create a commit not using gitextensions to see if you have the same behavior...

**Comment by pmme** [comment]
FWIW, I can reproduce this behavior by pressing SHIFT-ENTER instead of ENTER in the commit message.
I don't habitually do this but I had a feeling I knew what the author might have been doing.

I'm using the following:
- Git Extensions 4.0.0.15569
- Build cba315df5366fca01240d93b6e5a5077fd5e6630
- Git 2.38.1.windows.1
- Microsoft Windows NT 10.0.19043.0
- .NET 6.0.11
- DPI 96dpi (no scaling)

 
@huntran-clgx how do you insert a "new line" in your commit message? Do you use SHIFT-ENTER?

**Comment by pmme** [comment]
Yes, inside GitExtensions if I use SHIFT-ENTER then it shows as an empty rectangle for each newline character.
I also confirmed, by copying the commit message from GitExtensions using "Copy commit info" and pasting into Notepad++ and viewing non-printing characters, that it showed as a VT in NP++, as per the OP's statement.

**Comment by vbjay** [comment]
Added 
![image](https://user-images.githubusercontent.com/3629489/202583706-943f0310-5729-4bd5-9481-1d46ac58f39f.png)
to the text changed

```
Text:this is a shift enter
this is an enter

RTF:{\rtf1\ansi\ansicpg1252\deff0\nouicompat\deflang1033{\fonttbl{\f0\fnil\fcharset0 Segoe UI;}{\f1\fnil Segoe UI;}}
{\colortbl ;\red0\green0\blue0;}
{\*\generator Riched20 10.0.19041}\viewkind4\uc1 
\pard\cf1\f0\fs18 this is a shift enter\line\par
this is an enter\par
\f1\par
}
```

**Comment by RussKie** [comment]
It could be either or a combination of both:
1. font,
2. RTB handling

**Comment by vbjay** [comment]
Confirmed shift enter in RTB = VT

![image](https://user-images.githubusercontent.com/3629489/202584196-622755e8-b3bd-4817-b5ea-7921be866adb.png)

**Comment by vbjay** [comment]
Tried rewording and typing shift enter.  Did not accept the input using ge editor

![image](https://user-images.githubusercontent.com/3629489/202584674-c0f6626b-ce24-4e39-8700-52875b264371.png)

**Comment by vbjay** [comment]
![image](https://user-images.githubusercontent.com/3629489/202590363-40a40b27-4af8-4cb4-9f02-5d3849b62567.png)
![image](https://user-images.githubusercontent.com/3629489/202590425-0e6603e2-f67d-4366-bc58-ecbed8012980.png)
![image](https://user-images.githubusercontent.com/3629489/202590489-2cd32e7a-6d9c-4878-b2c2-8e10fa38459d.png)
| Number | Character |
|--------|-----------|
| 73     | I         |
| 32     |           |
| 97     | a         |
| 109    | m         |
| 32     |           |
| 112    | p         |
| 114    | r         |
| 101    | e         |
| 115    | s         |
| 115    | s         |
| 105    | i         |
| 110    | n         |
| 103    | g         |
| 32     |           |
| 115    | s         |
| 104    | h         |
| 105    | i         |
| 102    | f         |
| 116    | t         |
| 32     |           |
| 101    | e         |
| 110    | n         |
| 116    | t         |
| 101    | e         |
| 114    | r         |
| 32     |           |
| 104    | h         |
| 101    | e         |
| 114    | r         |
| 101    | e         |
| 11     |           |
| 97     | a         |
| 110    | n         |
| 100    | d         |
| 32     |           |
| 73     | I         |
| 32     |           |
| 97     | a         |
| 109    | m         |
| 32     |           |
| 112    | p         |
| 114    | r         |
| 101    | e         |
| 115    | s         |
| 115    | s         |
| 105    | i         |
| 110    | n         |
| 103    | g         |
| 32     |           |
| 101    | e         |
| 110    | n         |
| 116    | t         |
| 101    | e         |
| 114    | r         |
| 32     |           |
| 104    | h         |
| 101    | e         |
| 114    | r         |
| 101    | e         |
| 10     |           |
| 97     | a         |
| 110    | n         |
| 100    | d         |
| 32     |           |
| 116    | t         |
| 104    | h         |
| 105    | i         |
| 115    | s         |
| 32     |           |
| 105    | i         |
| 115    | s         |
| 32     |           |
| 110    | n         |
| 101    | e         |
| 120    | x         |
| 116    | t         |
| 32     |           |
| 108    | l         |
| 105    | i         |
| 110    | n         |
| 101    | e         |

```vb
Sub Main
	Dim frm As New Form
	frm.Size = New Size(500, 500)

	Dim rtb As New RichTextBox
	rtb.Dock = DockStyle.Fill
	rtb.Text = "I am pressing shift enter hereand I am pressing enter hereand this is next line"

	frm.Controls.Add(rtb)
	frm.ShowDialog
	rtb.Text.Dump("Text")
	Dim encodings = {
	System.Text.Encoding.Default} ',
	'System.Text.Encoding.UTF8,
	'System.Text.Encoding.Unicode,
	'System.Text.Encoding.UTF32


	Dim values = encodings.Select(Function(e) New With {.Bytes = e.GetBytes(rtb.Text), .Name = e.EncodingName, .CharNumbers = rtb.Text.Select(Function(c) New With {.Number = AscW(c), .Character = c})})

	values.Select(Function(v) New With {.CharNumbers = v.CharNumbers}).Dump("Characters") ', .Hex = String.Join("", v.Bytes.Select(Function(b) b.ToString("X2"))), .Name = v.Name, }).Dump("Encoding values")

End Sub

```
![image](https://user-images.githubusercontent.com/3629489/202591217-ba80d8bc-d115-4a5e-b53f-b6cba987093f.png)

**Comment by RussKie** [comment]
....what if (just if) we changed the RTB in the commit dialog with the ICSharpCode.TextEditor editor we use here?
![image](https://user-images.githubusercontent.com/4403806/202648432-b75e7022-1951-4159-9156-3fcc235ce87d.png)

**Comment by vbjay** [comment]
That's a huge lift.  I don't think it's needed. The op typed a different character and got a different result. He now knows not to not hit shift enter. There is a ton of functionality layered on top of the rtb that would need to be reworked.

**Comment by RussKie** [comment]
Makes sense.

**Comment by huntran-clgx** [comment]
Shift + Enter is more a habit since I don't want to accidentally commit something by pressing Enter and have the focus on the commit button by default.
It good to know the reason, thanks to @pmme
